lib/src/arbiter.dart

```diff
@@ -60,3 +60,84 @@ class PriorityArbiter extends Arbiter {
     ]);
   }
 }
+
+/// Round Robin Arbiter.
+class RoundRobinArbiter extends Arbiter {
+  /// Mask to define pending requests to be attended
+  /// the main idea is to mask the requests as they are granted to ensure all
+  /// requests are granted once before resetting the mask
+  /// example:
+  /// [clk] [_requests] [_requestMask]   [_grantMask]   [_grants]
+  ///  0    00100010   &  11111111      =   00100010  ->  00000010
+  ///  1    00100010   &  11111100      =   00100000  ->  00100000
+  ///  2    10000001   &  11000000      =   00100000  ->  10000000
+  /// On clock 0 and 1 the grants are used to generate the
+  /// requestMask, therefore grantMask changes to grant the next request
+  /// Clock 3 shows when a request changes, the requestMask make sure to
+  /// handle the missing requests before resetting
+  List<Logic> _requestMask = [];
+
+  /// Result of masking [_requests] with [_requestMask]
+  /// Contains bits of [_requests] but as bits are granted, will turn off
+  List<Logic> _grantMask = [];
+
+  /// Round Robin arbiter handles requests keeping track of previous requests.
+  RoundRobinArbiter(super.requests,
+      {required Logic clk, required Logic reset}) {
+    clk = addInput('clk', clk);
+    reset = addInput('reset', reset);
+    _requestMask = List.generate(count, (i) => Logic(name: 'requestMask$i'));
+    _grantMask = List.generate(count, (i) => Logic(name: 'grantMask$i'));
+    Sequential(clk, [
+      If(reset, then: [
+        for (var g = 0; g < count; g++) _requestMask[g] < 1,
+      ], orElse: [
+        // Use [_grants] to turn the bit granted to 0 along with all bits
+        // to the right to avoid, in case if  [_requests] changes, granting a
+        // previous request
+        // Example:
+        // [_grants] [requestMask]
+        // 00001000   11110000
+        CaseZ(_grants.rswizzle(), [
+          for (var i = 0; i < count; i++)
+            CaseItem(
+                Const(LogicValue.filled(count, LogicValue.z)
```

Comment:
I think you can set them to 0 instead of `z` since only one grant will ever be set?